protos/icmp/icmp.go

```diff
@@ -0,0 +1,329 @@
+package icmp
+
+import (
+	"net"
+	"time"
+
+	"github.com/elastic/libbeat/common"
+	"github.com/elastic/libbeat/logp"
+	"github.com/elastic/libbeat/publisher"
+
+	"github.com/elastic/packetbeat/config"
+	"github.com/elastic/packetbeat/protos"
+
+	"github.com/tsg/gopacket/layers"
+	// TODO: better use this? (also see commit #16691f7)
+	// "github.com/google/gopacket/layers"
+)
+
+type Processor interface {
+	ProcessICMPv4(tcphdr *layers.ICMPv4, pkt *protos.Packet)
+	ProcessICMPv6(tcphdr *layers.ICMPv6, pkt *protos.Packet)
+}
+
+// Contains all used information from the ICMP message on the wire.
+type IcmpMessage struct {
```

Comment:
Don't export symbols if not required. You might want to reduce some stuttering by renaming "IcmpMessage" to "message".